### DISCUSSION TITLE
Testing 1.21.0-beta1 mathematical expression with negative numbers

### DISCUSSION THREAD

**Discussion by UUoocl** [WarmUpTill/SceneSwitcher#704] [Q&A] [answered]
Thank you for the super useful plug-in. 
The new release 1.21.0-beta 1 worked with my previous macros on Windows 11.  
The new math expression worked with positive numbers, but negative numbers returned an error.  
In my test, a variable named "neg1" stores a negative number "-1".  The "neg1" variable is referenced by another variable set to mathematical expression. The error "failed evaluate expression "-1+1": is returned. 
![image](https://user-images.githubusercontent.com/99063397/228705298-4cb70ae8-b846-4c6c-bada-702030951f86.png)

Is it possible to calculate a mathematical expression that includes a negative number?

Thank you

**Comment by WarmUpTill** [comment, marked as answer]
I will look into it and report back if it is something can be implemented easily :)

Update:
Instead of trying to reinvent the wheel I now settled on using the "exprtk" library.
This should resolve your issues.

A build with this change should be available here in a few minutes:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/4579054873?pr=703

Let me know if you run into any issues! :)

**Reply by UUoocl** [reply]
Thank you very much! This update solved the issue.  
I tried some of the functions available in the "exprtk" library, and those worked as expected too.

This is a fantastic update that will enable new creative options.  Well done!
